cwltool/docker.py

```diff
@@ -309,7 +309,7 @@ def create_runtime(self, env, runtimeContext):
         if runtimeContext.rm_container:
             runtime.append(u"--rm")
 
-        runtime.append(u"--env=TMPDIR=/tmp")
+        runtime.append(u"--env=TMPDIR=%s" % self.builder.tmpdir)
```

Comment:
This should be a path within the container; `self.tmpdir` is mounted to `/tmp` at https://github.com/common-workflow-language/cwltool/pull/814/files#diff-616bb24f4b86bb222b6d286980bd2373L274

Reply:
Ah ok, `runtime.tmpdir` must be set incorrectly then

Reply:
I think the problem is actually at https://github.com/common-workflow-language/cwltool/blob/977962eca49fc126b839e8965e3f08240027bd0a/cwltool/process.py#L626 - paths that are passed into arguments to docker are resolved in the host environment, so `/tmp` gets translated to `/private/tmp` which doesn't match up with what we've specified as `TMPDIR`. I'll commit the changes that fix this